SofaKernel/modules/SofaSimulationCore/src/sofa/simulation/MechanicalOperations.h

```diff
@@ -162,6 +162,16 @@ class SOFA_SIMULATION_CORE_API MechanicalOperations
     void setDf(core::MultiVecDerivId& v);
     void setDf(core::ConstMultiVecDerivId& v);
 
+    /// Warn the user that a linear solver is required but has not been found
+    void showMissingLinearSolverError() const;
+    /// Create the list of all the available linear solver components as a string
+    static std::string listLinearSolverComponents();
+    /// Find all the available linear solver components
+    static type::vector<std::string> findLinearSolverComponentTypes();
```

Comment:
Not sure MechanicalOperations is the good entry point for this methods.
What about having them in ObjectFactory or somewhere in SofaBaseLinearSolver?

Reply:
Very good idea. There are plenty of places where it could be used. I put it in ObjectFactory